### DISCUSSION TITLE
Object syntax for other CSS properties?

### DISCUSSION THREAD

**Discussion by severinlandolt** [tailwindlabs/tailwindcss#11233] [General]
Hey everyone! I am wondering if it's true, that the object syntax only works for colors in the tailwind config?

Example of color object syntax here: [https://tailwindcss.com/docs/customizing-colors#color-object-syntax](https://tailwindcss.com/docs/customizing-colors#color-object-syntax)

I want to do something like this:

```diff
boxShadow: {
+      customname: {
          sm: "0 1px 2px 0 rgb(0 0 0 / 0.05)",
          DEFAULT: "0 1px 3px 0 rgb(0 0 0 / 0.1), 0 1px 2px -1px rgb(0 0 0 / 0.1)",
          md: "0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1)",
          lg: "0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)",
        },
      },
```

**Comment by wongjn** [comment]
The object syntax only works for colors in Tailwind configuration. This is because for the core plugins that use colors, they convert the object into a flat dictionary:

https://github.com/tailwindlabs/tailwindcss/blob/0e171fd2ddf578cd8cb30b12c64c14146414797e/src/corePlugins.js#L2139

However, core plugins that do not use colors don't do not expect a variable depth object of values and so they do not flatten the values:

https://github.com/tailwindlabs/tailwindcss/blob/0e171fd2ddf578cd8cb30b12c64c14146414797e/src/corePlugins.js#L2342

**Reply by severinlandolt** [reply]
Thank you!
